tldw_Server_API/app/api/v1/endpoints/admin/admin_byok.py

```
@@ -0,0 +1,107 @@
+    principal: AuthPrincipal = Depends(get_auth_principal),
+) -> SharedProviderKeyTestResponse:
+    return await admin_byok_service.test_shared_key(principal, payload)
+
+
+@router.get(
+    "/keys/shared",
+    response_model=SharedProviderKeysResponse,
+    dependencies=[Depends(require_roles("admin")), Depends(check_rate_limit)],
+)
+async def admin_list_shared_byok_keys(
+    principal: AuthPrincipal = Depends(get_auth_principal),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)